src/app/blog/iroh-content-discovery/page.mdx

```diff
@@ -0,0 +1,255 @@
+is already known, you can just skip it. You might also keep a cache of trackers
+and ask them for content before even asking the mainline DHT.
+
+The just acts as a last resort to find a tracker in case either you don't know
+a tracker or the tracker you have been using goes offline.
+
+## Signed announces
+
+Iroh nodes that share content have an ed keypair as the node id. So when
+announcing content we can sign the announce with this keypair. By doing this we
+prevent malicious actors from publishing fake announces to DDOS innocent nodes
+or to overwhelm the storage of the tracker.
+
+## Rate limiting
+
+The tracker needs to do strict rate limiting based on both ip address and node
+id. Rate limiting by node id is useful, but not sufficient. An adversary could
+just create a large number of node ids. By also limiting by ip address, we make
+sure that only a large botnet could DDOS a tracker.
+
+We can make DDOSing a tracker even harder by validating announces immediately
+after getting them, and then discarding all announces that have not ever been
+validated.
+
+## Validating announces
+
+Compared to other content-addressed systems, iroh has the advantage that BLAKE3
+allows verified requests for slices with the granularity of 1 KiB. So when a
+node claims that they host some content (and only the node, the owner of the ed
+private key, can make that claim), we can *probe* that node by requesting a
+random chunk of content.
+
+If the node does not exist, or if the node does not have that data, the request
+will fail, and we can treat the announce as invalid and discard it.
+
+It might be a good idea to require at least one successful probe before even
+storing the announce. This would be a further way to reduce the work to be done
+by the tracker in case of a fake announce. The worst an attacker could do is
+to force the tracker to do futile connection attempts to non-existing node ids.
+
+## Requesting content
+
+At this time, the content discovery system lives outside the blobs layer of iroh.
+When you want to download some content identified by a hash, you first ask the
+discovery system for a set of candidate nodes. In case of popular content
+you would ask for a random sample of candidate nodes.
+
+You then take this set of candidate nodes and tell iroh to download the data
+from any of these nodes.
+
+In the future we might integrate these two layers to have a pleasant API where
+you can just ask for the data for a hash, and iroh will handle both discovery
+and the actual download in an efficient matter.
+
+## Possible high level API
+
+An ideal API would probably look like this: You have a simple function to
+discover and retrieve content. This function returns a struct that can be either
+used as a stream to get all the nitty gritty details of content discovery and
+download progress, or used as a future that can just be awaited once the data
+is downloaded.
+
+# Current state 
```

Comment:
- should also have a future/what we are planning to do section, so folks understand where this is going